nameko/web/server.py

```diff
@@ -76,9 +76,21 @@ def run(self):
             sock, addr = self._sock.accept()
             sock.settimeout(self._serv.socket_timeout)
             self.container.spawn_managed_thread(
-                partial(self._serv.process_request, (sock, addr))
+                partial(self.process_request, sock, addr)
             )
 
+    def process_request(self, sock, address):
+        try:
+            self._serv.process_request((sock, address))
+        except OSError as exc:
+            # OSError("raw readinto() returned invalid length")
+            # can be raised when a client disconnects very early as a result
+            # of an eventlet bug: https://github.com/eventlet/eventlet/pull/353
+            # See https://github.com/onefinestay/nameko/issues/368
+            if "raw readinto() returned invalid length" in str(exc):
+                return
```

Comment:
I don't think so. The managed thread will exit without error and the server will continue waiting for another `accept()`